src/schema-association-service.ts

```diff
@@ -237,11 +252,61 @@ async function autoDetectSchema(
     if (session === undefined) {
         logger.log(`No organization found for ${workspaceFolder.name}`, 'SchemaDetection');
         vscode.window.showErrorMessage(format(Messages.unableToAccessOrganization, organizationName));
+        await delete1ESPTSchemaFileIfPresent(context);
         return undefined;
     }
 
     // Create the global storage folder to guarantee that it exists.
     await vscode.workspace.fs.createDirectory(context.globalStorageUri);
+    
+    logger.log(`Retrieving schema for ${workspaceFolder.name}`, 'SchemaDetection');
+
+    // Try to fetch schema in the following order:
+    // 1. Cached 1ESPT schema
+    // 2. 1ESPT schema if user is signed in with microsoft account and has enabled 1ESPT schema
+    // 3. Cached Organization specific schema
+    // 4. Organization specific schema
+    const token = await session.credentials2.getToken();
+    const authHandler = azdev.getBearerHandler(token.accessToken);
+    const azureDevOpsClient = new azdev.WebApi(`https://dev.azure.com/${organizationName}`, authHandler);
+
+    if (repoId1espt === undefined) {
```

Comment:
This if statement can be replaced by `repoId1espt ??= await get1ESPTRepoIdIfAvailable(azureDevOpsClient, organizationName);` (nullish coalescing).

Actually...shouldn't repoId1espt be _defined_ here? `const repoId1espt = ...`?

Reply:
Yes, we are caching the repoId1espt value. Will change it to let.